fms/modules/feedforward.py

```diff
@@ -267,8 +273,16 @@ def __init__(
             use_bias,
             gain,
         )
-        self.rank = rank
-        self.world_size = world_size
+        TPModule.setup_tp(self, rank, world_size)
+
+    def list_colwise_weights(self) -> List[str]:
+        return ["w1", "wg"]
+
+    def list_rowwise_weights(self) -> List[str]:
+        return ["w2"]
+
+    def list_embedding_weights(self) -> List[str]:
+        return []
```

Comment:
should we make these not abstract in the base class? then we'd only need to reimplement the ones we're actually changing.

Reply:
much cleaner, agree